overlord/snapstate/snapmgr.go

```diff
@@ -797,6 +797,11 @@ func (m *SnapManager) StartUp() error {
 	// the set of inhibited snaps is changed.
 	m.changeCallbackID = m.state.AddChangeStatusChangedHandler(processInhibitedAutoRefresh)
 
+	if CheckExpectedRestart(m.state) == ErrUnexpectedRuntimeRestart {
```

Comment:
can we have a simple unit test for this code in the package ? the actual complex logic is checked in managers_test.go but we could check the basics here as well for the error returning case

Reply:
There's `TestCheckExpectedRestartFromStartUpRequestsStop` in snapsate_test.go which checks StartUp()